src/routing_table/network_tests.rs

```diff
@@ -89,12 +92,24 @@ impl Network {
         }
     }
 
+    fn store_merge_info<T: PartialEq + Debug>(merge_info: &mut HashMap<Prefix<u64>, T>,
+                                              prefix: Prefix<u64>,
+                                              new_info: T) {
+        if let Some(content) = merge_info.get(&prefix) {
+            assert_eq!(new_info, *content);
+            return;
+        }
+        let _ = merge_info.insert(prefix, new_info);
+    }
+
+    // TODO: remove this when https://github.com/Manishearth/rust-clippy/issues/1279 is resolved
+    #[allow(for_kv_map)]
```

Comment:
That needs to be `#[cfg_attr(feature="clippy", allow(for_kv_map))]` to make it work without the `clippy` feature, too.